### PR TITLE
cod 0.1.0 (new formula)

### PR DESCRIPTION
**Pull request by iloveitaly**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/cod.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Cod < Formula
+  desc "Completion daemon for bash/fish/zsh"
+  homepage "https://github.com/dim-an/cod"
+  url "https://github.com/dim-an/cod/archive/9096c4bd0b4a34121f35c323e174a139c30aa8f6.tar.gz"
+  sha256 "dfc8de77d33d3a4e4af46eda3b61e4dbbeddf099d8f85bfc0b4c4a2d693b425c"
```
Comment (carlocab) [inline Formula/cod.rb:5]:
```suggestion
  url "https://github.com/dim-an/cod/archive/v0.0.0.tar.gz"
  sha256 "6f773358f363141dcd1bbdffaaa5dea0a40ae7a84d5fc55fd453302de8493c20"
```
Let's use the tarball from a release tag.

File: Formula/cod.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "make", "build"
+    bin.install "cod"
```
Comment (carlocab) [inline Formula/cod.rb:12]: Is there no `make install`?

File: Formula/cod.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    system bin/"cod", "--help"
```
Comment (carlocab) [inline Formula/cod.rb:16]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Comment by carlocab** [comment]
Thanks for your PR @iloveitaly. Just a few comments.

File: Formula/cod.rb (hunk 3 above)
Comment (iloveitaly) [reply Formula/cod.rb:16]: Good call! Will do.

File: Formula/cod.rb (hunk 1 above)
Comment (iloveitaly) [reply Formula/cod.rb:5]: Right now the release tarball is pretty out of date. I have an issue open on cod to add new releases and the maintainer is working on it there.

Should we wait to merge this until he has a new release out?

Comment (carlocab) [reply Formula/cod.rb:5]: We only build software from tagged releases, so we either use the existing one or wait for a newer tag.

Comment (iloveitaly) [reply Formula/cod.rb:5]: Got it, let's wait for a new tag then. Fine if we keep this PR open while the issue on the cod repo is resolved?

Comment (carlocab) [reply Formula/cod.rb:5]: Fine by me.

File: Formula/cod.rb (hunk 2 above)
Comment (iloveitaly) [reply Formula/cod.rb:12]: There is, but it installs the binary into a specific bin folder on the machine. `build` simply compiles the binary.

**Comment by iloveitaly** [comment]
@carlocab all fixed up! Mind taking another look?

File: Formula/cod.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "go" => :build
+
+  def install
+    system "make", "build"
+    bin.install "cod"
+  end
+
+  test do
+    assert_match "", shell_output("cod daemon")
+    assert_match "", shell_output("cod remove cod")
```
Comment (SMillerDev) [inline Formula/cod.rb:17]: Matching nothing seems kinda useless

Comment (iloveitaly) [reply Formula/cod.rb:17]: My assumption here is `shell_output` will throw an exception / fail if a non-zero exit code is encountered. Is that true?

I can change these to `asset(shell_output(...)).blank?` which is more what I'm looking for here, assuming exit codes are checked.

Comment (gromgit) [reply Formula/cod.rb:17]: `shell_output()` takes an optional return code to match against: https://rubydoc.brew.sh/Homebrew/Assertions.html#shell_output-instance_method

Comment (SMillerDev) [reply Formula/cod.rb:17]: > I can change these to asset(shell_output(...)).blank? which is more what I'm looking for here, assuming exit codes are checked.

Yes they are, please do

**Pull request closed without merging**